weblog/requesthandler.go

```diff
@@ -7,26 +7,29 @@ import (
 	"strconv"
 	"strings"
 
-	"github.com/deis/logger/syslogish"
+	"github.com/deis/logger/logs"
+	"github.com/deis/logger/storage"
 	"github.com/gorilla/mux"
 )
 
 type requestHandler struct {
-	syslogishServer *syslogish.Server
+	storageAdapter storage.Adapter
 }
 
-func newRequestHandler(syslogishServer *syslogish.Server) *requestHandler {
-	return &requestHandler{syslogishServer: syslogishServer}
+func newRequestHandler(storageAdapter storage.Adapter) *requestHandler {
```

Comment:
You should pass a pointer. If you don't, what you are receiving here and then holding a reference to is a _copy_ of a `storage.Adapter` instance. As the composition of the ring buffers in the _original_ `storage.Adapter` instance changes over time, this copy will _not_, thus retrieval of logs will not really work.